### ISSUE TITLE
'SonosData' object has no attribute 'entities'

### ISSUE THREAD

**Issue by alexives** [jeremywillans/sonos-crossfade#3]
### Background
First, I want to say thanks for making this integration. I super appreciate being able to change this setting!

I've been encountering this error when I try to enable or disable crossfade on one of our speakers as a part of a script. Seems to happen with both enabling and disabling crossfade. Based on some sleuthing, it looks like this might be related to the changes in https://github.com/home-assistant/core/pull/49441/

It looks like `SonosData` now links to `media_player_entities` rather than just `entities`. Also, instead of an array of entity objects, it's dict of RINCON_id -> entity object.

### Potentiall Fix?

I got it working by changing:
```
    def get_soco_from_entity_id(self, entity_id):
        """Returns the SoCo object of the entity that corresponds to the given entity_id"""
        data_sonos = self.hass.data[sonos.media_player.DATA_SONOS]

        for entity in data_sonos.entities:
            if entity.entity_id == entity_id:
                break
        else:
            entity = None

        if entity is None:
            _LOGGER.error("No entity found for entity_id: %s", entity_id)
            return None

        return entity.soco
```
To:
```
    def get_soco_from_entity_id(self, entity_id):
        """Returns the SoCo object of the entity that corresponds to the given entity_id"""
        data_sonos = self.hass.data[sonos.media_player.DATA_SONOS]

        for entity in data_sonos.media_player_entities.items():
            if entity[1].entity_id == entity_id:
                break
        else:
            entity = None

        if entity is None:
            _LOGGER.error("No entity found for entity_id: %s, %s", entity_id, data_sonos.media_player_entities)
            return None

        return entity[1].soco
```

May want a solution that looks for both for backwards compatibility? Or I dunno.

### Some Details

I'm calling the component with:
```
    - service: sonos_crossfade.update
      data:
        entity_id: media_player.upstairs_bedroom
        enabled: true
```

```
Logger: homeassistant.components.websocket_api.http.connection
Source: custom_components/sonos_crossfade/__init__.py:34
Integration: Home Assistant WebSocket API (documentation, issues)
First occurred: 10:55:33 PM (6 occurrences)
Last logged: 10:59:06 PM

[2704255632] 'SonosData' object has no attribute 'entities'
[2704255632] Error handling message: Unknown error
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 167, in handle_call_service
    await hass.services.async_call(
  File "/usr/src/homeassistant/homeassistant/core.py", line 1481, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1516, in _execute_service
    await handler.job.target(service_call)
  File "/usr/src/homeassistant/homeassistant/components/script/__init__.py", line 278, in service_handler
    await script_entity.async_turn_on(
  File "/usr/src/homeassistant/homeassistant/components/script/__init__.py", line 382, in async_turn_on
    await coro
  File "/usr/src/homeassistant/homeassistant/components/script/__init__.py", line 404, in _async_run
    return await self.script.async_run(variables, context)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1209, in async_run
    await asyncio.shield(run.async_run())
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 345, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 363, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 563, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1481, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1520, in _execute_service
    await self._hass.async_add_executor_job(handler.job.target, service_call)
  File "/usr/local/lib/python3.8/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/sonos_crossfade/__init__.py", line 51, in update
    device = self.get_soco_from_entity_id(entity_id)
  File "/config/custom_components/sonos_crossfade/__init__.py", line 34, in get_soco_from_entity_id
    for entity in data_sonos.entities:
AttributeError: 'SonosData' object has no attribute 'entities'
```

**Comment by dwradcliffe** [comment]
Same exact error here, starting with `2021.5` core release.
Looks like @alexives has done more digging than I have but I'm sure it's the same thing.

EDIT: The fix above works for me too.

**Comment by jeremywillans** [comment]
Thanks for the investigation @alexives !

Fixed in 0.3 release whilst maintaining legacy support 👍
